extensions/windows-event-log/ConsumeWindowsEventLog.cpp

```diff
@@ -0,0 +1,322 @@
+const std::string ConsumeWindowsEventLog::ProcessorName("ConsumeWindowsEventLog");
+
+static const int DEFAULT_MAX_BUFFER = 1024 * 1024;
+
+core::Property ConsumeWindowsEventLog::Channel(
+  core::PropertyBuilder::createProperty("Channel")->
+  isRequired(true)->
+  withDefaultValue("System")->
+  withDescription("The Windows Event Log Channel to listen to.")->
+  supportsExpressionLanguage(true)->
+  build());
+
+core::Property ConsumeWindowsEventLog::Query(
+  core::PropertyBuilder::createProperty("Query")->
+  isRequired(true)->
+  withDefaultValue("*")->
+  withDescription("XPath Query to filter events. (See https://msdn.microsoft.com/en-us/library/windows/desktop/dd996910(v=vs.85).aspx for examples.)")->
+  supportsExpressionLanguage(true)->
+  build());
+
+core::Property ConsumeWindowsEventLog::MaxBufferSize(
+  core::PropertyBuilder::createProperty("Max Buffer Size")->
+  isRequired(true)->
+  withDefaultValue(std::to_string(1024 * 1024))->
+  withDescription(
+    "The individual Event Log XMLs are rendered to a buffer."
+    " This specifies the maximum size in bytes that the buffer will be allowed to grow to. (Limiting the maximum size of an individual Event XML.)")->
+  build());
+
+core::Property ConsumeWindowsEventLog::InactiveDurationToReconnect(
+  core::PropertyBuilder::createProperty("Inactive Duration To Reconnect")->
+  isRequired(true)->
+  withDefaultValue("10 mins")->
+  withDescription(
+    "If no new event logs are processed for the specified time period, "
+    " this processor will try reconnecting to recover from a state where any further messages cannot be consumed."
+    " Such situation can happen if Windows Event Log service is restarted, or ERROR_EVT_QUERY_RESULT_STALE (15011) is returned."
+    " Setting no duration, e.g. '0 ms' disables auto-reconnection.")->
+  build());
+
+core::Relationship ConsumeWindowsEventLog::Success("success", "Relationship for successfully consumed events.");
+
+ConsumeWindowsEventLog::ConsumeWindowsEventLog(const std::string& name, utils::Identifier uuid)
+  : core::Processor(name, uuid), logger_(logging::LoggerFactory<ConsumeWindowsEventLog>::getLogger()) {
+  char buff[MAX_COMPUTERNAME_LENGTH + 1];
+  DWORD size = sizeof(buff);
+  if (GetComputerName(buff, &size))
+  {
+    computerName_ = buff;
+  }
+  else
+  {
+    LogWindowsError();
+  }
+}
+
+void ConsumeWindowsEventLog::initialize() {
+  //! Set the supported properties
+  setSupportedProperties({Channel, Query, MaxBufferSize, InactiveDurationToReconnect});
+
+  //! Set the supported relationships
+  setSupportedRelationships({Success});
+}
+
+void ConsumeWindowsEventLog::onSchedule(const std::shared_ptr<core::ProcessContext> &context, const std::shared_ptr<core::ProcessSessionFactory> &sessionFactory) {
+  if (subscriptionHandle_) {
+    logger_->log_error("Processor already subscribed to Event Log, expected cleanup to unsubscribe.");
+  } else {
+    subscribe(context);
+  }
```

Comment:
You brought up a good case in notifyStop. You should likely take a copy of the process session factory instead of session_, and create a new session when a stop notification arises.